api/src/org/pentaho/platform/api/engine/IServerStatusProvider.java

```diff
@@ -37,4 +43,33 @@
   public enum ServerStatus {
     DOWN, STARTING, STARTED, STOPPING, ERROR
   }
+
+  Locator LOCATOR = new Locator();
+
+  class Locator {
+    private static final String PROVIDER_CLASS = "org.pentaho.platform.api.engine.IServerStatusProvider.class";
+    IServerStatusProvider instance;
+    private static Logger logger = LoggerFactory.getLogger( IServerStatusProvider.Locator.class );
+
+    public IServerStatusProvider getProvider() {
+
+      if ( instance != null ) {
+        return instance;
+      }
+      if ( System.getProperty( PROVIDER_CLASS ) != null ) {
+
+        try {
+          instance = (IServerStatusProvider) Class.forName( System.getProperty( PROVIDER_CLASS ) ).newInstance();
+        } catch ( ClassNotFoundException e ) {
+          logger.error( "ServerStatusProvider class not found", e );
+        } catch ( InstantiationException | IllegalAccessException e ) {
+          logger.error( "ServerStatusProvider class could not be instantiated", e );
+        }
+        return instance;
```

Comment:
If any error occurs using their custom class we should instantiate and return the default